test/src/test/java/org/corfudb/integration/LogReplicationAbstractIT.java

```diff
@@ -323,6 +293,169 @@ private void triggerSnapshotAndTestRemainingEntries() throws Exception{
+                assertThat(map.count()).isEqualTo(numWrites);
+            }
+
+            // Confirm data does not exist on Standby Cluster
+            for(Table<Sample.StringKey, Sample.IntValueTag, Sample.Metadata> map : mapNameToMapStandby.values()) {
+                assertThat(map.count()).isZero();
+            }
+
+            startLogReplicatorServers();
+
+            // Verify Standby Status Listener received all expected updates (is_data_consistent)
+            log.info(">> Wait ... Replication status UPDATE ...");
+            statusUpdateLatch.await();
+
+            verifyDataOnStandby(numWrites);
+
+            // Confirm replication status reflects snapshot sync completed
+            log.info(">> Verify replication status completed on active");
+            verifyReplicationStatusFromActive();
+
+            verifySnapshotSync(standbyListener, totalStandbyStatusUpdates);
+
+            //reset the listener
+            standbyListener.reset(new CountDownLatch(totalStandbyStatusUpdates));
+
+
+            LogReplicationMetadata.ReplicationEventKey key = LogReplicationMetadata.ReplicationEventKey
+                    .newBuilder().setKey(UUID.randomUUID().toString()).build();
+            LogReplicationMetadata.ReplicationEvent event = LogReplicationMetadata.ReplicationEvent.newBuilder()
+                    .setClusterId(DefaultClusterConfig.getStandbyClusterId())
+                    .setEventId(UUID.randomUUID().toString())
+                    .setType(LogReplicationMetadata.ReplicationEvent.ReplicationEventType.FORCE_SNAPSHOT_SYNC)
+                    .build();
+
+            // add a valid snapshot sync event in the event table
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                txn.putRecord(replicationEventTable, key, event, null);
+                txn.commit();
+            }
+
+            standbyListener.countDownLatch.await();
+            verifySnapshotSync(standbyListener, totalStandbyStatusUpdates);
+
+            while(replicationEventTable.count() != 0) {
+                //wait
+            }
+            // verify the event table is cleared after an event is processed
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                assertThat(txn.getTable(REPLICATION_EVENT_TABLE_NAME).count()).isZero();
+            }
+
+            //reset the listener
+            standbyListener.reset(new CountDownLatch(totalStandbyStatusUpdates));
+
+            // Add an invalid entry to the table so the event listener's onError() gets called in a loop, giving the test to CP/trim
+            try(TxnContext txn = corfuStoreActive.txn(LogReplicationMetadataManager.NAMESPACE)) {
+                txn.putRecord(replicationEventTable, key, LogReplicationMetadata.ReplicationEvent.newBuilder().build(), null);
```

Comment:
will there be a failure in the listener itself or will snapshot sync eventually fail because the value is uninitialized?

Reply:
the failure is in listener thread, while it tries to enqueue the event in the discoveryServices's queue.
Basically on `OnNext()` -> `processEvent()`

Reply:
unable to follow how it will fail.. because everything in the event protobuf will have default values, including the event type and it will get added to the queue.  Am I missing something?

Reply:
yes, but it fails while trying to convert to UUID. This is the error with the stack trace:


```
2024-04-08T17:45:23.312Z | ERROR | treamPollingScheduler-worker-0 | r.c.StreamListenerResumePolicy | Exception caught during streaming processing. Re-subscribe this listener to latest timestamp. Error=
java.lang.IllegalArgumentException: Invalid UUID string:
        at java.base/java.util.UUID.fromString(UUID.java:215)
        at org.corfudb.infrastructure.logreplication.infrastructure.DiscoveryServiceEvent.<init>(DiscoveryServiceEvent.java:35)
        at org.corfudb.infrastructure.logreplication.infrastructure.LogReplicationEventListener.processEvent(LogReplicationEventListener.java:96)
        at org.corfudb.infrastructure.logreplication.infrastructure.LogReplicationEventListener.onNext(LogReplicationEventListener.java:85)
        at org.corfudb.runtime.collections.StreamListenerResumePolicy.onNextEntry(StreamListenerResumePolicy.java:75)
        at org.corfudb.runtime.collections.StreamListenerResumeOrFullSync.onNextEntry(StreamListenerResumeOrFullSync.java:20)
        at org.corfudb.runtime.collections.streaming.StreamingTask.lambda$produce$4(StreamingTask.java:194)
        at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:79)
        at org.corfudb.common.metrics.micrometer.MicroMeterUtils.time(MicroMeterUtils.java:115)
        at org.corfudb.runtime.collections.streaming.StreamingTask.lambda$produce$5(StreamingTask.java:194)
        at java.base/java.util.Optional.ifPresent(Optional.java:183)
        at org.corfudb.runtime.collections.streaming.StreamingTask.produce(StreamingTask.java:194)
        at org.corfudb.runtime.collections.streaming.StreamingTask.run(StreamingTask.java:223)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
```

Reply:
So the listener is gets resubscribed and then again hits this...keeping it in this loop until we run CP/trim twice

Reply:
got it.. can you add this in the comments or test description (something like 'event id is empty so construction of DiscoveryServiceEvent throws an exception')?  This is for the reader to be sure that the resubscription will indeed happen.